api/v1alpha1/argocd_conversion.go

```diff
@@ -0,0 +1,577 @@
+	// deprecated keycloak configs set in alpha (.spec.sso.image, .spec.sso.version, .spec.sso.verifyTLS, .spec.sso.resources),
+	// override .spec.sso.keycloak in beta
+	if src.Spec.SSO != nil && !reflect.DeepEqual(src.Spec.SSO, &ArgoCDSSOSpec{}) {
+		if src.Spec.SSO.Image != "" || src.Spec.SSO.Version != "" || src.Spec.SSO.VerifyTLS != nil || src.Spec.SSO.Resources != nil {
+			if sso.Keycloak == nil {
+				sso.Keycloak = &v1beta1.ArgoCDKeycloakSpec{}
+			}
+			sso.Keycloak.Image = src.Spec.SSO.Image
+			sso.Keycloak.Version = src.Spec.SSO.Version
+			sso.Keycloak.VerifyTLS = src.Spec.SSO.VerifyTLS
+			sso.Keycloak.Resources = src.Spec.SSO.Resources
+		}
+	}
+
+	// deprecated dex configs set in alpha (.spec.dex), override .spec.sso.dex in beta
+	if src.Spec.Dex != nil && !reflect.DeepEqual(src.Spec.Dex, &ArgoCDDexSpec{}) && (src.Spec.Dex.Config != "" || src.Spec.Dex.OpenShiftOAuth) {
+		if sso == nil {
+			sso = &v1beta1.ArgoCDSSOSpec{}
+		}
+		sso.Provider = v1beta1.SSOProviderTypeDex
+		sso.Dex = (*v1beta1.ArgoCDDexSpec)(src.Spec.Dex)
+	}
+
+	dst.Spec.SSO = sso
+
+	// rest of the fields
+	dst.Spec.ApplicationSet = ConvertAlphaToBetaApplicationSet(src.Spec.ApplicationSet)
+	dst.Spec.ExtraConfig = src.Spec.ExtraConfig
+	dst.Spec.ApplicationInstanceLabelKey = src.Spec.ApplicationInstanceLabelKey
+	dst.Spec.ConfigManagementPlugins = src.Spec.ConfigManagementPlugins
+	dst.Spec.Controller = *ConvertAlphaToBetaController(&src.Spec.Controller)
+	dst.Spec.DisableAdmin = src.Spec.DisableAdmin
+	dst.Spec.ExtraConfig = src.Spec.ExtraConfig
+	dst.Spec.GATrackingID = src.Spec.GATrackingID
+	dst.Spec.GAAnonymizeUsers = src.Spec.GAAnonymizeUsers
+	dst.Spec.Grafana = *ConvertAlphaToBetaGrafana(&src.Spec.Grafana)
+	dst.Spec.HA = *ConvertAlphaToBetaHA(&src.Spec.HA)
+	dst.Spec.HelpChatURL = src.Spec.HelpChatURL
+	dst.Spec.HelpChatText = src.Spec.HelpChatText
+	dst.Spec.Image = src.Spec.Image
+	dst.Spec.Import = (*v1beta1.ArgoCDImportSpec)(src.Spec.Import)
+	dst.Spec.InitialRepositories = src.Spec.InitialRepositories
+	dst.Spec.InitialSSHKnownHosts = v1beta1.SSHHostsSpec(src.Spec.InitialSSHKnownHosts)
+	dst.Spec.KustomizeBuildOptions = src.Spec.KustomizeBuildOptions
+	dst.Spec.KustomizeVersions = ConvertAlphaToBetaKustomizeVersions(src.Spec.KustomizeVersions)
+	dst.Spec.OIDCConfig = src.Spec.OIDCConfig
+	dst.Spec.Monitoring = v1beta1.ArgoCDMonitoringSpec(src.Spec.Monitoring)
+	dst.Spec.NodePlacement = (*v1beta1.ArgoCDNodePlacementSpec)(src.Spec.NodePlacement)
+	dst.Spec.Notifications = v1beta1.ArgoCDNotifications(src.Spec.Notifications)
+	dst.Spec.Prometheus = *ConvertAlphaToBetaPrometheus(&src.Spec.Prometheus)
+	dst.Spec.RBAC = v1beta1.ArgoCDRBACSpec(src.Spec.RBAC)
+	dst.Spec.Redis = v1beta1.ArgoCDRedisSpec(src.Spec.Redis)
+	dst.Spec.Repo = v1beta1.ArgoCDRepoSpec(src.Spec.Repo)
+	dst.Spec.RepositoryCredentials = src.Spec.RepositoryCredentials
+	dst.Spec.ResourceCustomizations = src.Spec.ResourceCustomizations
```

Comment:
aren't we dropping `Spec.ResourceCustomizations` in the `v1beta1` spec as well?

Reply:
Yes. I will get those changes in a separate PR.